fdbserver/VersionedBTree.actor.cpp

```diff
@@ -1436,15 +1817,61 @@ class DWALPager : public IPager2 {
 				    .detail("DesiredPageSize", self->desiredPageSize);
 			}
 
+			self->setExtentSize(self->pHeader->pagesPerExtent);
+
 			self->freeList.recover(self, self->pHeader->freeList, "FreeListRecovered");
+			self->extentFreeList.recover(self, self->pHeader->extentFreeList, "ExtentFreeListRecovered");
 			self->delayedFreeList.recover(self, self->pHeader->delayedFreeList, "DelayedFreeListRecovered");
+			self->extentUsedList.recover(self, self->pHeader->extentUsedList, "ExtentUsedListRecovered");
 			self->remapQueue.recover(self, self->pHeader->remapQueue, "RemapQueueRecovered");
 
-			Standalone<VectorRef<RemappedPage>> remaps = wait(self->remapQueue.peekAll());
-			for (auto& r : remaps) {
-				self->remappedPages[r.originalPageID][r.version] = r.newPageID;
+			debug_printf("DWALPager(%s) Queue recovery complete.\n", self->filename.c_str());
+
+			// remapQueue entries are recovered using a fast path reading extents at a time
+			// we first issue disk reads for remapQueue extents obtained from extentUsedList
+			Standalone<VectorRef<ExtentUsedListEntry>> extents = wait(self->extentUsedList.peekAll());
+			debug_printf("DWALPager(%s) ExtentUsedList size: %d.\n", self->filename.c_str(), extents.size());
+			if (extents.size() > 1) {
+				QueueID remapQueueID = self->remapQueue.queueID;
+				for (int i = 1; i < extents.size() - 1; i++) {
+					if (extents[i].queueID == remapQueueID) {
+						LogicalPageID extID = extents[i].extentID;
+						debug_printf("DWALPager Extents: ID: %s ", toString(extID).c_str());
+						self->readExtent(extID);
+					}
+				}
 			}
 
+			// And here we consume results of the disk reads and populate the remappedPages map
+			// Using a promiseStream for the peeked results ensures that we use the CPU to populate the map
+			// and the disk concurrently
+			state PromiseStream<Standalone<VectorRef<RemappedPage>>> remapStream;
+			state Future<Void> remapRecoverActor;
+			remapRecoverActor = self->remapQueue.peekAllExt(remapStream);
+			state int remapEntriesRead = 0;
+			loop choose {
```

Comment:
Apply same rewrite here as the other peekAll reader.

Reply:
Done